src/lib/src/grid3d/cell.cpp

```diff
@@ -99,4 +103,126 @@ cell_corners(const size_t i,
     return corners;
 }
 
+std::vector<double>
+get_corners_minmax(std::vector<double> &corners)
+{
+    double xmin = std::numeric_limits<double>::max();
+    double xmax = std::numeric_limits<double>::min();
+    double ymin = std::numeric_limits<double>::max();
+    double ymax = std::numeric_limits<double>::min();
+    double zmin = std::numeric_limits<double>::max();
+    double zmax = std::numeric_limits<double>::min();
+    for (auto i = 0; i < 24; i += 3) {
+        if (corners[i] < xmin) {
+            xmin = corners[i];
+        }
+        if (corners[i] > xmax) {
+            xmax = corners[i];
+        }
+        if (corners[i + 1] < ymin) {
+            ymin = corners[i + 1];
+        }
+        if (corners[i + 1] > ymax) {
+            ymax = corners[i + 1];
+        }
+        if (corners[i + 2] < zmin) {
+            zmin = corners[i + 2];
+        }
+        if (corners[i + 2] > zmax) {
+            zmax = corners[i + 2];
+        }
+    }
+    std::vector<double> minmax{ xmin, xmax, ymin, ymax, zmin, zmax };
+    return minmax;
+}  // get_corners_minmax
+
+/*
+ * Estimate if a point is inside a cell face top (option != 1) or cell face bottom
+ * (option = 1), seen from above, and return True if it is inside, False otherwise.
+ * @param x X coordinate of the point
+ * @param y Y coordinate of the point
+ * @param i The (i) coordinate
+ * @param j The (j) coordinate
+ * @param k The (k) coordinate
+ * @param corners A vector of doubles, length 24
+ * @param point A vector of doubles, length 2
+ * @return Boolean
+ */
+bool
+is_xy_point_in_cell(const double x,
+                    const double y,
+                    const size_t i,
+                    const size_t j,
+                    const size_t k,
+                    const size_t ncol,
+                    const size_t nrow,
+                    const size_t nlay,
+                    const py::array_t<double> &coordsv,
+                    const py::array_t<float> &zcornsv,
+                    const int option = 0)
+{
+
+    double crn[8][3]{};
```

Comment:
Consider making this a std::vector<std::vector, n>, m>

Reply:
I would think std::array<std::array<double, 3>, 8> crn{}; is better since this a fixed size array